online-workflow/src/main/java/eu/itesla_project/online/db/OnlineDbMVStore.java

```diff
@@ -1002,16 +1018,16 @@ public void storeState(String workflowId, Integer stateId, Network network) {
             Properties parameters = new Properties();
             parameters.setProperty("iidm.export.xml.indent", "true");
             parameters.setProperty("iidm.export.xml.with-branch-state-variables", "true");
-            parameters.setProperty("iidm.export.xml.with-breakers", "true");
-            parameters.setProperty("iidm.export.xml.with-properties", "true");
             Exporters.export("XIIDM", network, parameters, dataSource);
             // store network state values, for later serialization
-            Map<HistoDbAttributeId, Object> networkValues = IIDM2DB.extractCimValues(network, new IIDM2DB.Config(network.getId(), true, true)).getSingleValueMap();
-            ConcurrentHashMap<Integer, Map<HistoDbAttributeId, Object>> workflowStates = new ConcurrentHashMap<Integer, Map<HistoDbAttributeId, Object>>();
-            if (workflowsStates.containsKey(workflowId))
-                workflowStates = workflowsStates.get(workflowId);
-            workflowStates.put(stateId, networkValues);
-            workflowsStates.put(workflowId, workflowStates);
+            if (contingencyId == null) {
+                Map<HistoDbAttributeId, Object> networkValues = IIDM2DB.extractCimValues(network, new IIDM2DB.Config(network.getId(), true, true)).getSingleValueMap();
+                ConcurrentHashMap<Integer, Map<HistoDbAttributeId, Object>> workflowStates = new ConcurrentHashMap<Integer, Map<HistoDbAttributeId, Object>>();
+                if (workflowsStates.containsKey(workflowId))
+                    workflowStates = workflowsStates.get(workflowId);
+                workflowStates.put(stateId, networkValues);
+                workflowsStates.put(workflowId, workflowStates);
```

Comment:
no it isn't:  in your suggested code the statement 
`workflowStates.put(stateId, networkValues); `
would be useless ( workflowStates vs. workflowsStates ) ...